src/hypertable.c

```diff
@@ -1048,7 +1048,7 @@ hypertable_chunk_store_add(const Hypertable *h, const Chunk *input_chunk)
 	ts_subspace_store_add(h->chunk_cache,
 						  cached_chunk->cube,
 						  cached_chunk,
-						  /* object_free = */ NULL);
+						  (void (*)(void *)) ts_chunk_free);
```

Comment:
That's a bit of a risky cast. Why not do it inside the function like in `/src/nodes/chunk_dispatch.c`:
```
static void
destroy_chunk_insert_state(void *cis)
{
	ts_chunk_insert_state_destroy((ChunkInsertState *) cis);
}
```

Reply:
You mean, in case the signature changes? Sure, I can do it.

Reply:
Technically it is considered undefined behaviour. I'm sure it works in practice e.g. for gcc.